cmd/clusterctl/test/e2e/move_test.go

```diff
@@ -0,0 +1,171 @@
+		initTestMgmtCluster(ctx, toMgmtInfo)
+
+		// Do the move
+		c, err := clusterctlclient.New(fromMgmtInfo.clusterctlConfigFile)
+		Expect(err).ToNot(HaveOccurred())
+		err = c.Move(clusterctlclient.MoveOptions{
+			FromKubeconfig: fromMgmtInfo.mgmtCluster.KubeconfigPath,
+			ToKubeconfig:   toMgmtInfo.mgmtCluster.KubeconfigPath,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+	}, setupTimeout)
+
+	AfterEach(func() {
+		fmt.Fprintf(GinkgoWriter, "Tearing down kind clusters\n")
+		fromMgmtInfo.mgmtCluster.Teardown(ctx)
+		toMgmtInfo.mgmtCluster.Teardown(ctx)
+		fmt.Fprintf(GinkgoWriter, "Tearing down kind workload cluster\n")
+		if err := cluster.NewProvider().Delete(workloadInfo.workloadClusterName, ""); err != nil {
+			// Treat this as a non critical error
+			fmt.Fprintf(GinkgoWriter, "Deleting the kind cluster %q failed. You may need to remove this by hand.\n", workloadInfo.workloadClusterName)
+		}
+	})
+
+	Context("single node workerload cluster", func() {
+		It("should move all Cluster API objects to the new mgmt cluster, unpause the Cluster and delete all objects from previous mgmt cluster", func() {
+			Eventually(
+				func() error {
+					if err := toMgmtInfo.mgmtClient.Get(ctx, client.ObjectKey{Namespace: "default", Name: workloadInfo.workloadClusterName}, &clusterv1.Cluster{}); err != nil {
+						return err
+					}
+					if err := toMgmtInfo.mgmtClient.Get(ctx, client.ObjectKey{Namespace: "default", Name: "controlplane-0"}, &clusterv1.Machine{}); err != nil {
+						return err
+					}
+					if err := toMgmtInfo.mgmtClient.Get(ctx, client.ObjectKey{Namespace: "default", Name: "controlplane-0-config"}, &bootstrapv1.KubeadmConfig{}); err != nil {
+						return err
+					}
+					if err := toMgmtInfo.mgmtClient.Get(ctx, client.ObjectKey{Namespace: "default", Name: workloadInfo.workloadClusterName}, &infrav1.DockerCluster{}); err != nil {
+						return err
+					}
+					if err := toMgmtInfo.mgmtClient.Get(ctx, client.ObjectKey{Namespace: "default", Name: "controlplane-0"}, &infrav1.DockerMachine{}); err != nil {
+						return err
+					}
+					return nil
+				}, 3*time.Minute, 5*time.Second,
+			).ShouldNot(HaveOccurred())
+			// Should unpause Cluster object in the new mgmt cluster.
+			Eventually(
+				func() (bool, error) {
+					testCluster := &clusterv1.Cluster{}
+					if err := toMgmtInfo.mgmtClient.Get(ctx, client.ObjectKey{Namespace: "default", Name: workloadInfo.workloadClusterName}, testCluster); err != nil {
+						return false, err
+					}
+					if testCluster.Spec.Paused {
+						return false, nil
+					}
+					return true, nil
+				}, 3*time.Minute, 5*time.Second,
+			).Should(BeTrue())
+			// Should delete all Cluster API objects from the previous management cluster.
+			Eventually(
+				func() error {
+					err := fromMgmtInfo.mgmtClient.Get(ctx, client.ObjectKey{Namespace: "default", Name: workloadInfo.workloadClusterName}, &clusterv1.Cluster{})
+					if err == nil || !apierrors.IsNotFound(err) {
```

Comment:
should these all be `err != nil && !apierrors.IsNotFound(err)`?

Reply:
See earlier comment.